tests/search.py

```diff
@@ -76,6 +152,31 @@ def _test_unit_text_search(qs, text, sfields, exact, empty=True):
             assert searchword_found
 
 
+def _test_units_state_filter(qs, unit_filter):
+    result = UnitStateFilter(qs).filter(unit_filter)
+    for item in result:
+        assert item in qs
+    assert result.count() == result.distinct().count()
```

Comment:
the result would differ if the query didnt give back a unique qs - in this case thats unlikely but not impossible.